client/src/components/PreviewIcon.module.css

```diff
@@ -0,0 +1,31 @@
+.preview-icon-default {
+    display: flex;
+    flex-direction: column;
+    gap: 1em;
+    color: var(--text-medium);
+}
+
+    .preview-icon-explorer {
+        composes: preview-icon-default;
+        height: 192px;
+        width: 120px;
+        border: none;
+        background-color: white;
+        padding: 1rem;
+        padding-bottom: 1rem;
+    }
+
+        .preview-icon-explorer-title {
+            font-weight: bold;
+        }
+
+        .preview-icon-explorer-body {
+            text-align: left;
```

Comment:
So I did some research and it seems that adding ellipses to multi-line overflows is one of those unsolved CSS mysteries.
This solution might look obscure but to me it seems the most elegant and flexible way as it avoids any euristics. As of 2023 this is supported by every popular browser.